src/lib.rs

```diff
@@ -17,6 +17,38 @@ pub struct AutoInheritConf {
     pub prefer_simple_dotted: bool,
 }
 
+/// Rewrites a `path` dependency as being absolute, based on a given path
+fn rewrite_dep_paths_as_absolute<'a, P: AsRef<std::path::Path>>(
+    deps: impl Iterator<Item = &'a mut Dependency>,
+    parent: P,
+) {
+    deps.for_each(|dep| {
+        if let Dependency::Detailed(detail) = dep {
+            detail.path = detail.path.as_mut().map(|path| {
+                parent
+                    .as_ref()
+                    .join(path)
+                    .canonicalize()
+                    .unwrap()
+                    .to_string_lossy()
+                    .to_string()
+            })
+        }
+    });
+}
+
+/// Rewrites a `path` dependency as being relative, based on a given path
+fn rewrite_dep_path_as_relative<P: AsRef<std::path::Path>>(dep: &mut Dependency, parent: P) {
+    if let Dependency::Detailed(detail) = dep {
+        detail.path = detail.path.as_mut().map(|path| {
+            pathdiff::diff_paths(path, parent.as_ref())
+                .unwrap()
+                .to_string_lossy()
```

Comment:
We should never use `to_string_lossy` @hdoordt @cbgbt.
We should fail if the path is not UT8-encoded, but we shouldn't lose fidelity.